freqtrade/freqtradebot.py

```diff
@@ -126,6 +131,34 @@ def _throttle(self, func: Callable[..., Any], min_secs: float, *args, **kwargs)
         time.sleep(duration)
         return result
 
+    def refresh_tickers(self, pair_list: List[str]) -> Dict:
+        """
+        Refresh tickers asyncronously and return the result.
+        """
+        # TODO: maybe add since_ms to use async in the download-script?
+        # TODO: only refresh once per interval ? *may require this to move to freqtradebot.py
+        # TODO: Add tests for this and the async stuff above
+         
+        ticker_interval = self.strategy.ticker_interval
+        interval_in_seconds = constants.TICKER_INTERVAL_MINUTES[ticker_interval] * 60
+
+        should_not_update = ((self._klines_last_fetched_time + interval_in_seconds +1) > round(time.time()))
+
+        if should_not_update:
+            return False
+
+        logger.debug("Refreshing klines for %d pairs", len(pair_list))
+        datatups = asyncio.get_event_loop().run_until_complete(
+            self.exchange.async_get_candles_history(pair_list, ticker_interval))
+
+        # fetching the timestamp of last candle
+        self._klines_last_fetched_time = datatups[0][1][-1][0] / 1000
```

Comment:
i think we maybe should move the "should_not_update" logic to the async function and always call get_candles_history.
should one pair fail to refresh, we would not refresh it for `ticker_interval` as long as the first pair did refresh.
@misaghshakeri what do you think?

Reply:
it shouldn't create a lock/race condition (async is not threads), it's still running in sequence (somehow) ...

be carefull with `_cached_ticker` - we have 2 different datatypes, one for history (klines) - which we are "asyncing" now - and one for the "last tick" - which "_cached_ticker" is for, but it's only used by RPC functions (for which it does not really matter if the data is a minute out of date) - but not for anything trade-related... at least not as far as i am aware of